### PR TITLE
Change github markdown templates to yaml

### PR DESCRIPTION
**Pull request by ahlag**
## What changes are proposed in this pull request?
Change github markdown templates to yaml

## How is this patch tested?
Tested Locally on a different forked repository
https://github.com/ahlag/mlflow

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by ahlag** [comment]
Getting the following error in Release-note category / validate (pull_request). Can the maintainer apply the category label?
```
Release note category labels applied to this PR:
[]
Error: No release note category label is applied to this PR. Please select a checkbox in the release note category section: https://github.com/mlflow/mlflow/pull/5803#user-content-release-note-category or manually apply a release note category label (e.g. 'rn/bug-fix') if you're a maintainer of this repository. If this job failed when a release note category label is already applied, please re-run it.
```

**Comment by harupy** [comment]
Can we rebase on master?

**Comment by harupy** [comment]
> Getting the following error in Release-note category / validate (pull_request). Can the maintainer apply the category label?

I reran the job. It should pass now.

File: .github/ISSUE_TEMPLATE/feature_request_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/feature_request_template.yaml:10]:
```suggestion
        Thank you for submitting a feature request. **Before proceeding, please review MLflow's [Issue Policy for feature requests](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md#feature-requests) and the [MLflow Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst)**.
```

File: .github/ISSUE_TEMPLATE/feature_request_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/feature_request_template.yaml:37]: Let's add `value` that looks like:

```
### What is the use case for this feature?

### Why is this use case valuable to support for MLflow users in general?

### Why is this use case valuable to support for your project(s) or organization?

### Why is it currently difficult to achieve this use case? (please be as specific as possible about why related MLflow features and components are insufficient)
```

File: .github/ISSUE_TEMPLATE/installation_issue_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/installation_issue_template.yaml:16]:
```suggestion
        Describe the system where you encountered the installation.
```

File: .github/ISSUE_TEMPLATE/installation_issue_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/installation_issue_template.yaml:32]:
```suggestion
        ```bash
        pip install mlflow=x.y.z
        ```
```

File: .github/ISSUE_TEMPLATE/installation_issue_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/installation_issue_template.yaml:53]:
```suggestion
        ```
        ERROR: Could not find a version that satisfies the requirement mlflow==x.y.z
        ```
```

File: .github/ISSUE_TEMPLATE/feature_request_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/feature_request_template.yaml:110]:
```suggestion
      required: false
```

File: .github/ISSUE_TEMPLATE/feature_request_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/feature_request_template.yaml:27]:
```suggestion
        In a few sentences, provide a clear, high-level description of the feature request
```

File: .github/ISSUE_TEMPLATE/feature_request_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/feature_request_template.yaml:108]:
```suggestion
        Use this section to include any additional information about the feature. If you have a proposal for how to implement this feature, please include it here. For implementation guidelines, please refer to the [Contributing Guide](https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#contribution-guidelines).
```

**Comment by ahlag** [comment]
@harupy 
Not sure why after rebasing on master, my previous commits from bug-reports are still visible here.
screen_shot 2022-04-30 at 20 45 55

**Comment by harupy** [comment]
@ahlag Can you try this https://stackoverflow.com/a/25357146?

```bash
git reset $(git merge-base master $(git branch --show-current))
git add -A
git commit -s -m "<commit message>"
```

**Comment by ahlag** [comment]
@harupy 
Thank you for helping!

I have tried with some luck.
```console
❯ git reset $(git merge-base master $(git branch --show-current))                                                                 ─╯
Unstaged changes after reset:
D	.github/ISSUE_TEMPLATE/bug_report_template.md
D	.github/ISSUE_TEMPLATE/doc_fix_template.md
D	.github/ISSUE_TEMPLATE/feature_request_template.md
D	.github/ISSUE_TEMPLATE/installation_issue_template.md
M	mlflow/server/js/src/experiment-tracking/components/HtmlTableView.js
M	mlflow/server/js/src/experiment-tracking/components/MetricView.css
M	mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.js
M	mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.test.js
M	mlflow/server/js/src/experiment-tracking/reducers/MetricReducer.js
M	mlflow/server/js/src/experiment-tracking/reducers/MetricReducer.test.js
M	mlflow/server/js/src/experiment-tracking/reducers/Reducers.js
M	mlflow/server/js/src/i18n/default/en.json

❯ git add -A
❯ git commit -s -m "reset index"
[change-templates-to-yaml c06ba1fc] reset index
 18 files changed, 1343 insertions(+), 230 deletions(-)
 delete mode 100644 .github/ISSUE_TEMPLATE/bug_report_template.md
 create mode 100644 .github/ISSUE_TEMPLATE/bug_report_template.yaml
 delete mode 100644 .github/ISSUE_TEMPLATE/doc_fix_template.md
 create mode 100644 .github/ISSUE_TEMPLATE/doc_fix_template.yaml
 delete mode 100644 .github/ISSUE_TEMPLATE/feature_request_template.md
 create mode 100644 .github/ISSUE_TEMPLATE/feature_request_template.yaml
 delete mode 100644 .github/ISSUE_TEMPLATE/installation_issue_template.md
 create mode 100644 .github/ISSUE_TEMPLATE/installation_issue_template.yaml
 create mode 100644 mlflow/server/js/src/experiment-tracking/components/MetricsSummaryTable.js
 create mode 100644 mlflow/server/js/src/experiment-tracking/components/MetricsSummaryTable.test.js
 
❯ git push --force-with-lease origin change-templates-to-yaml                                                                     ─╯
To github.com:ahlag/mlflow.git
 ! [rejected]          change-templates-to-yaml -> change-templates-to-yaml (stale info)
error: failed to push some refs to 'github.com:ahlag/mlflow.git'
```

**Comment by harupy** [comment]
Have you tried `git push -f origin change-templates-to-yaml`?

**Comment by ahlag** [comment]
@harupy 
Ok done! Thanks!

**Comment by harupy** [comment]
@ahlag can you temporarily change the default branch of your fork to this PR branch?

**Comment by ahlag** [comment]
@harupy 
Ok done!

File: .github/ISSUE_TEMPLATE/feature_request_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/feature_request_template.yaml:114]: Can we movr this section after the motivation section?

File: .github/ISSUE_TEMPLATE/feature_request_template.yaml
Comment (harupy) [inline .github/ISSUE_TEMPLATE/feature_request_template.yaml:45]:
```suggestion
        #### What is the use case for this feature?

        #### Why is this use case valuable to support for MLflow users in general?

        #### Why is this use case valuable to support for your project(s) or organization?

        #### Why is it currently difficult to achieve this use case?
```

because the section header (`Motivation`) uses `###`.

**Review by harupy — APPROVED** [review]
LGTM once the remaining comments are addressed!

**Comment by ahlag** [comment]
@harupy 
Ok done!

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**
